### PR TITLE
Support custom lightning profilers in config

### PR DESCRIPTION
**Pull request by pbelevich**
# What does this PR do ?

This PR adds support of defining custom lightning profilers

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
To use `PyTorchProfiler` after 100 warm-up steps a user can define trainer.profiler as follows:
```
trainer:
  profiler:
    _target_: pytorch_lightning.profilers.PyTorchProfiler
    schedule:
      _target_: torch.profiler.schedule
      skip_first: 100
      wait: 5
      warmup: 1
      active: 3
      repeat: 2
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [x] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [x] Did you write any new necessary tests?
- [x] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [x] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?
@MaximumEntropy @ericharper @ekmb @yzhang123 @VahidooX @vladgets @malay-nagda @mikolajblaz @yaoyu-33 @akoumpa

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

**Comment by akoumpa** [comment]
Thanks @pbelevich, is this ready?

File: nemo/collections/nlp/parts/megatron_trainer_builder.py [hunk 1]
```diff
@@ -180,15 +182,30 @@ def _callbacks(self, callbacks: Optional[list]) -> list:
...
     def create_trainer(self, callbacks=None) -> Trainer:
         # cfg.trainer.precision becomes None in Trainer if precision_plugins exist since both precision plugins and precision
         precision = self.cfg.trainer.precision
         strategy = self._training_strategy()
         plugins = self._plugins()
         callbacks = self._callbacks(callbacks)
-        trainer = Trainer(plugins=plugins, strategy=strategy, **self.cfg.trainer, callbacks=callbacks)
+        # save cfg.trainer.profiler because it will be deleted
+        original_profiler = self.cfg.trainer.get('profiler', None)
+        profiler = self._profiler()
```
Comment (akoumpa) [inline nemo/collections/nlp/parts/megatron_trainer_builder.py:201]: Can I burden you with adding a comment explaining what `_profiler` does?
Also re. naming, I would like to include a verb, maybe `_reinstantiate_profiler`? but up to you.

Comment (pbelevich) [reply nemo/collections/nlp/parts/megatron_trainer_builder.py:218]: I added a doc string for `_profiler`. I named it `_profiler` to preserve existing method style: `_training_strategy`, `_grad_scaler`, `_plugins`, `_callbacks`

**Comment by pbelevich** [comment]
@akoumpa could you please run the CI?

**Comment by pbelevich** [comment]
@akoumpa I don't know why CI didn't run on this PR

**Comment by pbelevich** [comment]
@akoumpa could you please take a look?
